### PR TITLE
Use builder for ClusterStats

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStats.java [hunk 1]
```diff
@@ -13,168 +13,113 @@
...
-import static com.google.common.base.MoreObjects.toStringHelper;
+import java.util.Map;
 
-public class ClusterStats
+import static java.util.Objects.requireNonNull;
+
+public record ClusterStats(
+        String clusterId, int runningQueryCount,
+        int queuedQueryCount,
+        int blockedQueryCount,
```
Comment (willmostly) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStats.java:25]: `blockedQueryCount` is not used anywhere. It should correspond to `BasicQueryStats.isFullyBlocked()`, as mentioned in https://github.com/trinodb/trino-gateway/issues/233. 

I think we should remove it, as it doesn't seem like an important dimension of cluster health. What is your opinion?

**Review by willmostly — APPROVED** [review]
This looks like a good change. Ptal at my comment on the `blocked` field - just thought i'd note it in case it is more convenient to remove it here vs a  separate PR.

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStats.java (hunk 1 above)
Comment (ebyhr) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStats.java:25]: Good catch. Let's remove the field in a follow-up.

**Comment by ebyhr** [comment]
@wendigo @mosabua Gentle reminder.

**Review by mosabua — APPROVED** [review]
Looks good. 

Just our of curiosity .. are you aiming to update other record classes to the same pattern as well?

**Comment by ebyhr** [comment]
No. We don't need to change the existing recrod which don't require setters.

**Comment by mosabua** [comment]
> No. We don't need to change the existing record which don't require setters.

Cool.

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStats.java (hunk 1 above)
Comment (ebyhr) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStats.java:26]: https://github.com/trinodb/trino-gateway/pull/241

**Pull request merged**